### PR TITLE
feat(README): add why klog

### LINKED ISSUE #43: Why klog?
/kind bug

It would be nice for the README file for `klog` to explain why the fork, versus contributing to glog.

### PR DESCRIPTION
**Pull request by jaredallard**
**What this PR does / why we need it**: Adds context to why this repo was created

**Which issue(s) this PR fixes**: Fixes #43

**Special notes for your reviewer**: Took my first crack at it, LMK if it's missing the point entirely!

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,27 @@
 klog
 ====
 
-klog is a permanant fork of https://github.com/golang/glog. original README from glog is below
+klog is a permanant fork of https://github.com/golang/glog.
```
Comment (neolit123) [inline README.md]: `permanent`

Comment (jaredallard) [reply README.md]: Going to fix, but interestingly, that wasn't my typo :wink:

**Comment by jaredallard** [comment]
@yagonobre Done

File: README.md [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+## Why was klog created?
+
+The decision to create klog was one that wasn't made lightly, but it was necessary due to some
+drawbacks that are present in [glog](https://github.com/golang/glog). Ultimately, the fork was created due to glog not being under active development; this can be seen in the glog README:
+
+> The code in this repo [...] is not itself under development
+
+This makes us unable to solve many use cases without a fork. The factors that contributed to needing feature development are listed below:
+
+ * `glog` presents a lot "gotchas" that aren't well documented, not necessarily needed (specifically challenges that are present in container environments). This makes using `glog` in shared libraries somewhat dangerous and unpredictable
```
Comment (justinsb) [inline README.md]: I'm not sure what "not necessarily needed"  refers to here?

**Comment by justinsb** [comment]
Thanks @jaredallard - one question but otherwise LGTM.  I'm generally in favor of merging something rather than trying to merge the perfect thing, so I'd be OK with merging this, but it might be easier just to link "presented a lot of gotchas" to issue 61006.

File: README.md (hunk 2 above)
Comment (jaredallard) [reply README.md]: Yeah, honestly not sure what happened there. Will fix.

**Comment by jaredallard** [comment]
@justinsb Introduced changes based on your feedback.

**Comment by jaredallard** [comment]
Can we get some movement on this if anyone has time? 🎉

**Comment by dims** [comment]
/approve
/lgtm

Thanks @jaredallard

**Pull request merged**